packages/discord.js/src/structures/MessagePayload.js

```diff
@@ -148,11 +148,13 @@ class MessagePayload {
     }
 
     let flags;
-    if (this.isMessage || this.isMessageManager) {
+    if (typeof this.options.flags !== 'undefined' || this.isMessage || this.isMessageManager) {
       // eslint-disable-next-line eqeqeq
       flags = this.options.flags != null ? new MessageFlags(this.options.flags).bitfield : this.target.flags?.bitfield;
-    } else if (isInteraction && this.options.ephemeral) {
-      flags = MessageFlags.FLAGS.EPHEMERAL;
+    }
+
+    if (isInteraction && this.options.ephemeral) {
+      flags |= MessageFlags.FLAGS.EPHEMERAL;
```

Comment:
You can use -types `MessageFlag` here instead. I'll be removing the `FLAGS` prop from the bitfield in my other PR.